src/Http/Http.Abstractions/src/Extensions/RequestTrailerExtensions.cs

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.Extensions.Primitives;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNetCore.Http
+{
+    /// <summary>
+    /// HttpRequest extensions for working with request trailing headers.
+    /// </summary>
+    public static class RequestTrailerExtensions
+    {
+        /// <summary>
+        /// Gets the request "Trailer" header that lists which trailers to expect after the body.
+        /// </summary>
+        /// <param name="request"></param>
+        /// <returns></returns>
+        public static StringValues GetDeclaredTrailers(this HttpRequest request)
+        {
+            return request.Headers.GetCommaSeparatedValues(HeaderNames.Trailer);
+        }
+
+        /// <summary>
+        /// Indicates if the request supports receiving trailer headers.
+        /// </summary>
+        /// <param name="request"></param>
+        /// <returns></returns>
+        public static bool SupportsTrailers(this HttpRequest request)
+        {
+            return request.HttpContext.Features.Get<IHttpRequestTrailersFeature>() != null;
```

Comment:
This is now true even for requests that don't support request bodies and HTTP/1.1 Content-Length requeusts too?

I don't feel too strongly about this, but maybe `SupportsTrailers` should be a property on IHttpRequestTrailersFeature like `Available` is. That way SupportsTrailers could still be false despite IHttpRequestTrailersFeature despite being present on every request in Kestrel.

That being said, I think this is OK as is for now. We can add a default interface method later if customers complain the SupportsTrailers() extension method returns true too often.